src/TopDownProteomics/Proteomics/BrnoModificationLookup.cs

```diff
@@ -32,22 +47,81 @@ public IProteoformModification GetModification(ProFormaDescriptor descriptor)
 
             switch (abbreviation)
             {
-                case "ac": return new BrnoModification();
-                case "me1": return new BrnoModification();
-                case "me2s":
-                case "me2a":
-                case "me2": return new BrnoModification();
-                case "me3": return new BrnoModification();
-                case "ph": return new BrnoModification();
+                case "ac": return _modifications[0];
+                case "me1": return _modifications[1];
+                case "me2s": return _modifications[2];
+                case "me2a": return _modifications[3];
+                case "me2": return _modifications[4];
+                case "me3": return _modifications[5];
+                case "ph": return _modifications[6];
 
                 default:
-                    throw new ProteoformHypothesisCreateException($"Couldn't handle value for descriptor {descriptor.Key}:{descriptor.Value}.");
+                    throw new ProteoformGroupCreateException($"Couldn't handle value for descriptor {descriptor.Key}:{descriptor.Value}.");
             }
         }
 
-        private class BrnoModification : IProteoformModification
+        private BrnoModification[] CreateModificationArray(IElementProvider elementProvider)
         {
+            var mods = new BrnoModification[7];
+
+            var h = elementProvider.GetElement(1);
+            var c = elementProvider.GetElement(6);
+            var o = elementProvider.GetElement(8);
+            var p = elementProvider.GetElement(15);
 
+            mods[0] = new BrnoModification("ac", new[]
+            {
+                new EntityCardinality<IElement>(c, 2),
+                new EntityCardinality<IElement>(h, 2),
+                new EntityCardinality<IElement>(o, 1)
+            });
+            mods[1] = new BrnoModification("me1", new[]
+            {
+                new EntityCardinality<IElement>(c, 1),
+                new EntityCardinality<IElement>(h, 2)
+            });
+
+            var me2 = new[]
+            {
+                new EntityCardinality<IElement>(c, 2),
+                new EntityCardinality<IElement>(h, 4)
+            };
+            mods[2] = new BrnoModification("me2s", me2);
```

Comment:
I suppose it doesn't matter that these `me2` arrays will be at the same place in memory, since they become readonly down below in `BrnoModification`.

Reply:
Exactly, I like to save memory when I can. Most of the interfaces in the SDK (e.g. IElement, IChemicalFormula) are readonly to encourage the creation of value objects (http://enterprisecraftsmanship.com/2015/01/03/value-objects-explained/) and the use of the Flyweight pattern (https://en.wikipedia.org/wiki/Flyweight_pattern)